tutorials/io/ntuple/ntpl016_streaming_vector.C

```diff
@@ -0,0 +1,191 @@
+#include <ROOT/RNTupleTypes.hxx>
+#include <ROOT/RNTupleView.hxx>
+#include <ROOT/RNTupleWriter.hxx>
+
+#include <TRandom3.h>
+
+#include <cstdint>
+#include <iostream>
+#include <vector>
+#include <utility>
+
+constexpr char const *kFileName = "ntpl015_streaming_vector.root";
+constexpr char const *kNTupleName = "ntpl";
+constexpr char const *kFieldName = "LargeVector";
+constexpr unsigned int kNEvents = 10;
+constexpr unsigned int kVectorSize = 1000000;
+
+// Create an RNTuple with a single vector field. Every entry contains a large vector of random integers.
+// The vector should be seen as too large to be held entirely in memory during reading.
+void CreateRNTuple()
+{
+   auto model = ROOT::RNTupleModel::Create();
+   auto ptrLargeVector = model->MakeField<std::vector<std::uint32_t>>(kFieldName);
+   auto writer = ROOT::RNTupleWriter::Recreate(std::move(model), kNTupleName, kFileName);
+
+   auto prng = std::make_unique<TRandom3>();
+   prng->SetSeed();
+
+   for (ROOT::NTupleSize_t i = 0; i < kNEvents; i++) {
+      ptrLargeVector->clear();
+      for (std::size_t j = 0; j < kVectorSize; j++)
+         ptrLargeVector->emplace_back(prng->Integer(-1));
+      writer->Fill();
+   }
+   std::cout << "RNTuple written" << std::endl;
+}
+
+/*
+ * ==================================================================================================
+ */
+
+// For comparison, the canonical read function that reads the entire vector for every entry.
+void ReadRNTupleSimple()
+{
+   auto reader = ROOT::RNTupleReader::Open(kNTupleName, kFileName);
+
+   const auto nEntries = reader->GetNEntries();
+   std::cout << "Simple reading, found " << nEntries << " entries" << std::endl;
+
+   auto ptrLargeVector = reader->GetModel().GetDefaultEntry().GetPtr<std::vector<std::uint32_t>>(kFieldName);
+   for (ROOT::NTupleSize_t i = 0; i < nEntries; i++) {
+      reader->LoadEntry(i);
+
+      const auto vectorSize = ptrLargeVector->size();
+      uint64_t sum = 0;
+      for (auto val : *ptrLargeVector)
+         sum += val;
+
+      std::cout << "Size and sum of vector: " << vectorSize << " " << sum << std::endl;
+   }
+   std::cout << "RNTuple simple read" << std::endl;
+}
+
+/*
+ * ==================================================================================================
+ */
+
+// The StreamingVectorView class allows iteration over an RNTuple on-disk vector of element type T.
+// Unlike an std::vector, this class does not provide random-access but only allows to iterate the data elements
+// from beginning to end.
+// Internally, it uses an RNTupleCollection view and an item view to load chunks of the vector elements into memory,
+// so that never the entire vector needs to stay in memory.
+// Note that we don't need to implement loading chunks of data explicitly. Simply by asking for a single vector element
+// at every iteration step, the RNTuple views will take care of keeping only the currently required data pages
+// in memory.
```

Comment:
It would be helpful here to say how one can tune the maximum amount of memory used by the StreamingVector